dm/worker/server.go

```diff
@@ -676,15 +676,22 @@ func (s *Server) OperateMysqlWorker(ctx context.Context, req *pb.MysqlWorkerRequ
 		resp.Msg = errors.ErrorStack(err)
 		return resp, nil
 	}
-	if req.Op == pb.WorkerOp_UpdateConfig || req.Op == pb.WorkerOp_StopWorker {
-		w := s.stopWorker(cfg.SourceID)
-		if w == nil {
+	if req.Op == pb.WorkerOp_UpdateConfig {
+		w, err := s.stopWorker(cfg.SourceID)
+		if err != nil {
 			resp.Result = false
-			resp.Msg = "Mysql task has not been created, please call CreateMysqlTask. Or there has been a worker started" +
-				" which has different config"
+			resp.Msg = errors.ErrorStack(err)
 			return resp, nil
 		}
 		w.Close()
+	} else if req.Op == pb.WorkerOp_StopWorker {
+		w, err := s.stopWorker(cfg.SourceID)
+		if err == terror.ErrWorkerSourceNotMatch {
+			resp.Result = false
```

Comment:
How about setting `Msg` here too?

Reply:
Yes. I made a mistake, and I have fixed it.